diff --git a/source/ng/NewgroundsHandler.hx b/source/ng/NewgroundsHandler.hx
index 501ee0e..d6ca225 100644
--- a/source/ng/NewgroundsHandler.hx
+++ b/source/ng/NewgroundsHandler.hx
@@ -25,8 +25,7 @@ class NewgroundsHandler
 
 	public var medals:Map<String, MedalDef> = [];
 
-	public function new(use_medals:Bool = true, use_scoreboards:Bool = false, ?login_callback:Void->Void)
-		init(use_medals, use_scoreboards, login_callback);
+	public function new() {}
 
 	public function init(use_medals:Bool = true, use_scoreboards:Bool = false, ?login_callback:Void->Void)
 	{
@@ -62,20 +61,21 @@ class NewgroundsHandler
 		var app_id = json.app_id;
 		var encryption_key = json.encryption_key;
 
-		NG.createAndCheckSession(app_id, false);
+		NG.createAndCheckSession(app_id, false, null, on_session_created);
 		NG.core.setupEncryption(encryption_key, AES_128, BASE_64);
 
 		NG.core.onLogin.add(() -> onNGLogin(login_callback));
+	}
 
-		if (!NG.core.loggedIn)
-		{
-			trace("Waiting on manual login...");
-			NG.core.requestLogin();
-		}
-		else
+	function on_session_created(outcome:LoginOutcome)
+	{
+		if (NG.core.loggedIn)
 		{
-			NG_LOGGED_IN = true;
+			return;
 		}
+
+		trace("Waiting on manual login...");
+		NG.core.requestLogin();
 	}
 
 	function load_medal_defs()
diff --git a/source/states/LoadGameState.hx b/source/states/LoadGameState.hx
index 9002cc6..bb9e3fb 100644
--- a/source/states/LoadGameState.hx
+++ b/source/states/LoadGameState.hx
@@ -24,7 +24,8 @@ class LoadGameState extends BaseState
 		super();
 
 		#if newgrounds
-		Main.ng_api = new NewgroundsHandler(true, false, on_logged_in);
+		Main.ng_api = new NewgroundsHandler();
+		Main.ng_api.init(true, false, on_logged_in);
 		#else
 		on_logged_in();
 		#end
